spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -54,8 +58,10 @@ public ProducerDestination provisionProducerDestination(String name,
 	public ConsumerDestination provisionConsumerDestination(String name, String group,
 			ExtendedConsumerProperties<PubSubConsumerProperties> properties)
 			throws ProvisioningException {
+		// Use group name as subscription name
+		// Generate anonymous random group, if one not provided
+		String subscription = !StringUtils.hasText(group) ? "anonymous." + UUID.randomUUID().toString() : group;
```

Comment:
Just to clarify on the feature request, is there to be exactly 1 anonymous subscription for each topic that every subscriber of the anonymous group will use? Or will there be multiple anonymous subscriptions for each topic?

If only 1 anonymous subscription per topic, one option is you could not worry about deleting it and just leave it there; in subsequent application runs, you could check if the anonymous subscription already exists for the topic and reuse it if present or create if it does not exist.

Reply:
I believe anonymous subscriptions have to be volatile and die together with their subscriber.